src/components/FilterNavigation/FilterNavigation.js

```diff
@@ -8,29 +8,36 @@ import {
 
 import { segments } from '../../constants';
 
-const FilterNavigation = ({ segment, onChange }) => (
+const FilterNavigation = ({ segment, segmentsSortBy, onChange }) => (
   <ButtonGroup
     fullWidth
     data-test-filters-navigation
   >
     {
-      Object.keys(segments).map(name => (
-        <Button
-          key={`${name}`}
-          to={`/inventory?segment=${name}&sort=title`}
-          buttonStyle={`${segment === name ? 'primary' : 'default'}`}
-          id={`segment-navigation-${name}`}
-          onClick={onChange}
-        >
-          <FormattedMessage id={`ui-inventory.filters.${name}`} />
-        </Button>
-      ))
+      Object.keys(segments).map((name) => {
+        const sort = segmentsSortBy.find(x => x.name === name).sort || '';
+
+        return (
+          <>
+            <Button
+              key={`${name}`}
+              to={`/inventory?segment=${name}${sort !== '' ? '&sort=' : ''}${sort}`}
```

Comment:
Isn't the title default sort?

Reply:
Yes there is set in initial component state. Upper code with empty sort option is for relevance where sort param not exists in link location